### PR TITLE
devilutionX: update to 1.2.1

### PR DESCRIPTION
**Pull request by bugcrazy**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/devilutionX/template [hunk 1]
```diff
@@ -1,16 +1,16 @@
 # Template file for 'devilutionX'
 pkgname=devilutionX
-version=1.1.0
+version=1.2.1
 revision=1
 build_style=cmake
-configure_args="-DBINARY_RELEASE=ON -DTTF_FONT_PATH=\"/usr/share/fonts/truetype/CharisSILB.ttf\""
+configure_args="-DVERSION_NUM=1.2.1 -DBINARY_RELEASE=ON -DTTF_FONT_PATH=\"/usr/share/fonts/truetype/CharisSILB.ttf\""
```
Comment (ericonr) [inline srcpkgs/devilutionX/template:6]:
```suggestion
configure_args="-DBINARY_RELEASE=ON -DTTF_FONT_PATH=\"/usr/share/fonts/truetype/CharisSILB.ttf\"
 -DVERSION_NUM=$version"
```

**Comment by Chocimier** [comment]
This version cross builds.

File: srcpkgs/devilutionX/template (hunk 1 above)
Comment (bugcrazy) [reply srcpkgs/devilutionX/template]: This new argument is necessary for compilation, but each time changes appear in the tools they compile, forcing you to change the code in the templates!    diasurgical/devilutionX#1401

Comment (ericonr) [reply srcpkgs/devilutionX/template]: I am aware it's necessary, just moved it to a different line so the line didn't get too big.

**Comment by ericonr** [comment]
What Chocimier meant is that `nocross` should be removed.

And please squash all the commits together :)

File: srcpkgs/devilutionX/template (hunk 1 above)
Comment (bugcrazy) [reply srcpkgs/devilutionX/template]: I understand!

**Comment by bugcrazy** [comment]
Squash all the commits, done!

File: srcpkgs/devilutionX/template [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+configure_args="-DVERSION_NUM=$version -DBINARY_RELEASE=ON -DTTF_FONT_PATH=\"/usr/share/fonts/truetype/CharisSILB.ttf\""
 makedepends="SDL2-devel SDL2_ttf-devel SDL2_mixer-devel libsodium-devel"
 short_desc="Diablo I engine for modern operating systems"
 maintainer="MarcoAPC <marcoaureliopc@gmail.com>"
 license="Unlicense"
 homepage="https://github.com/diasurgical/devilutionX"
 distfiles="https://github.com/diasurgical/devilutionX/archive/${version}.tar.gz"
-checksum=395ec298df5383abe60550d45adba5e5811984589ccb1e504891b267b3c467c1
+checksum=002dcbd4d4a5bdf8db1a3ec01139e5bfbed46d6a1caa32b17c9f2df161ad3521
 nocross="https://build.voidlinux.org/builders/aarch64-musl_builder/builds/25453/steps/shell_3/logs/stdio"
```
Comment (ericonr) [inline srcpkgs/devilutionX/template:14]: Please remove this line.

Comment (bugcrazy) [reply srcpkgs/devilutionX/template]: Done!

**Pull request merged**